pkg/workloads/spark_job/spark_util.py

```diff
@@ -220,53 +229,61 @@ def ingest(ctx, spark):
         df = read_parquet(ctx, spark)
 
     input_type_map = {f.name: f.dataType for f in df.schema}
-
-    for raw_column_name in ctx.raw_columns.keys():
+    inferred_col_type_map = {c.name: c.dataType for c in df.schema}
+    for raw_column_name in ctx.raw_columns:
         raw_column = ctx.raw_columns[raw_column_name]
-        expected_types = CORTEX_TYPE_TO_ACCEPTABLE_SPARK_TYPES[raw_column["type"]]
-        actual_type = input_type_map[raw_column_name]
-        if actual_type not in expected_types:
-            logger.error("found schema:")
-            log_df_schema(df, logger.error)
-
-            raise UserException(
-                "raw column " + raw_column_name,
-                "type mismatch",
-                "expected {} but found {}".format(
-                    " or ".join(str(x) for x in expected_types), actual_type
-                ),
-            )
-        target_type = CORTEX_TYPE_TO_SPARK_TYPE[raw_column["type"]]
+        expected_cortex_type = raw_column["type"]
+        if expected_cortex_type == consts.COLUMN_TYPE_INFERRED:
+            column_type = SPARK_TYPE_TO_CORTEX_TYPE[inferred_col_type_map[raw_column_name]]
+            ctx.write_metadata(raw_column["id"], {"type": column_type})
+        else:
+            actual_spark_type = input_type_map[raw_column_name]
+            if actual_spark_type not in SPARK_TYPE_TO_CORTEX_TYPE:
+                actual_spark_type = StringType()
```

Comment:
Is this correct? If we get an unexpected type, we need to cast it to string, but setting `actual_spark_type` to `StringType()` means that we won't run the cast (lines 257-259). Can you add a test for this?